### ISSUE TITLE
how to get right result from addGather() with explict batch and dynamic shape?

### ISSUE THREAD

**Issue by dulvqingyun** [NVIDIA/TensorRT#660]
## Description

 when i using network->addGather(*network->getOutput(0),*filtered_score->getOutput(1),0) , where *network->getOutput(0) is a tensor with dims(-1, -1,-1) , *filtered_score->getOutput(1) is a tensor with dims(-1, 1000) when build network,the first dim is batch size。i get the output with dim(-1,1000,-1,-1) by  "roi_bbox_filtered->getOutput(0)->getDimensions()" but not dim(-1,1000,-1) as i expected.
after build the engine, i see the actual dim of tensor *network->getOutput(0) is (5,16368,4),  dim of tensor *filtered_score->getOutput(1) is (5,1000), and the gatherlayer dim is (5,1000,16368,4) but not (5,1000,4) as i expected.
i expected the output tenor dim number is 3, but got 4. anybody could help me ?

## Environment

**TensorRT Version*7.0*: 
**GPU Type*P4*: 
**Nvidia Driver Version*9.0*: 
**CUDA Version*9.0*: 
**CUDNN Version*7.6*: 
**Operating System + Version*ubuntu 16*: 
**Python Version (if applicable)**: 
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**:

**Comment by mk-nvidia** [comment]
FYI @kevinch-nv

**Comment by nvpohanh** [comment]
@dulvqingyun Could you try TRT 8.4 and see if the issue still exists? Thanks

**Comment by ttyio** [comment]
Close since no activation for more than 3 weeks, thanks!
